views/histogram.py

```diff
@@ -0,0 +1,68 @@
+import dash_html_components as html
+import dash_bootstrap_components as dbc
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+from datetime import timedelta
+
+from components import histogram
+from components.config import histogram_config
+from utils import earthquake_data
+from app import app
+
+
+def get_layout(session_id):
+    """ Return the layout for a histogram and its configurations.
+
+    Keyword arguments:
+    session_id -- ID of the current session
+    """
+    eq_data = earthquake_data.get_earthquake_data(session_id)
+
+    if eq_data is None or eq_data.data.shape == (0, 0):
+        return 'No uploaded data found'
+
+    start_date, end_date = eq_data.get_daterange()
+    default_end_date = start_date + timedelta(weeks=1)
+    filtered_data = eq_data.filter_by_dates(start_date, default_end_date)
+
+    default_size_column = eq_data.get_column_params(
+        eq_data.get_magnitudes().name
+    )
+
+    return html.Div([
+            dbc.Col(
+                html.Div(
+                    id='histogram',
+                    children=histogram.get_component(None)
+                )
```

Comment:
You could use the same class as in scatter plot, 'plot_sidebar_open', and add a similar callback that adjusts the size of the plot according to whether the sidebar is collapsed or not